### DISCUSSION TITLE
Redis 6.2 formulae disabled

### DISCUSSION THREAD

**Discussion by ljodal** [Homebrew/discussions#4592] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.24-26-g8d9aa90
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8d9aa9070a90b1b660ef1f2b0f8132ada4f4e45a
Last commit: 47 minutes ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 51565f0be5ebfdca2a944aff8f693ff2f7114b0a
Core tap last commit: 5 months ago
Core tap branch: master
Core tap JSON: 22 Jun 12:09 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 10
no_proxy: *
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 12.6.6-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  jsonschema
  redis@6.2
  swiftgen
```

### Description of issue

The redis@6.2 formula is disabled, but as far as I can tell it's still maintained upstream. [The PR that added the formula](https://github.com/Homebrew/homebrew-core/pull/102258), and set the deprecation date, seems to indicate that redis 6.2 will no longer be maintained [once Redis 8 is released](https://github.com/Homebrew/homebrew-core/pull/102258#issuecomment-1146802681). The last release to the 6.2 version was just two months ago and there's no redis version 8 released yet. The [Homebrew docs](https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae#disabling) seem to indicate that a package should be disabled if it hasn't been maintained for a long time, which does not seem to be the case here.

We're using the redis@6.2 package locally because that's the latest version supported by Google Memorystore for Redis, which is what we run in production. We'd thus like to continue using redis 6 locally and it would be nice to avoid having to maintain this formula ourselves.

I wasn't sure where to raise this. May I just open a PR that removes the disabling of the package?

**Comment by carlocab** [comment]
> May I just open a PR that removes the disabling of the package?

For small PRs, it's probably better to open them first then ask questions later. In this case, you should probably go ahead and do that, but please include a reference to upstream documentation indicating that redis 6.2 is still supported in your PR.

**Comment by SMillerDev** [comment]
Yeah, it seems development slowed and my prediction based on the docs didn't really come to pass. Instead of

> now: 7.0, 6.2, 6.0
> 6 months: 7.1, 7.0, 6.2
> 1 year: 8.0, 7.1, 7.0

It seems to be 

> now: 7.0, 6.2, 6.0
> 6 months: 7.0, 6.2, 6.0
> 1 year: (almost 7.2), 7.0, 6.2, 6.0

**Comment by ljodal** [comment]
Alright, thanks for the replies! I've opened a PR here https://github.com/Homebrew/homebrew-core/pull/134737 :)
